### DISCUSSION TITLE
clickhouseexporter otel tables

### DISCUSSION THREAD

**Discussion by ghost** [open-telemetry/opentelemetry-collector-contrib#22147] [Q&A] [answered]
it is really helpful to include the default table structure of ClickHouse for otel_logs, otel_traces, otel_metrics, below documentation says table names but no create statements.

https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/clickhouseexporter

I am doing POC on observability to push data to ClickHouse ,it is really helpful to know the standard table structure of this 3 tables.

**Comment by Frapschen** [comment, marked as answer]
The table schemas can find in:

- traces: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/exporter_traces.go#L162
- logs: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/exporter_logs.go#L120
- metrics, exponential_histogram: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/internal/exponential_histogram_metrics.go#L20
- metrics, gauge: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/internal/gauge_metrics.go#L20
- metrics, histogram: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/internal/histogram_metrics.go#L20
- metrics, sum: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/internal/sum_metrics.go#L20
- metrics, summary: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/internal/summary_metrics.go#L20
